docker/README.md

```diff
@@ -0,0 +1,24 @@
+# Docker Image Tutorial
```

Comment:
I'd suggest the tutorial here be focused on using the image available in docker hub (`rtohaan/alexandrie:latest`) with the default settings (eg `sqlite`, port `3000`, etc.), steps for how to set up each of the directories in a developer's local directory, and more importantly _why_ each of these directories matter.

I like calling out the dependencies required, and sharing the link to the Docker installation page.

To that end, I propose to refactor into the following structure:

---------------------------------------------------------

# Running Alexandrie via Docker

## Introduction

This tutorial walks through setting up a local instance of Alexandrie so you can test publishing a crate to an [alternative registry](https://doc.rust-lang.org/cargo/reference/registries.html#using-an-alternate-registry).

## Dependencies

- [Docker](https://docs.docker.com/get-docker/)

## Steps

### Pull Image

We provide a docker image you can use to run a local instance of Alexandrie. Pull it with 
```
docker pull rtohaan/alexandrie:latest
```
This image uses the default Alexandrie [configuration](../alexandrie.toml) and comes with `sqlite` installed.

Alternatively, you can build it yourself from the root of this repo with
```
docker build -t alexandrie -f Dockerfile .
```

The remaining steps assume you have pulled `rtohaan/alexandrie:latest`.

### Setting up directories for application data

_(notes like "Pick a clean directory. Here we will create the directories for Alexandrie to use and persist data.")_

_notes on setting up `./crate-storage` and what purpose it serves_

_notes on setting up `./data` and what purpose it serves_

_notes on setting up `./crate-index` and what purpose it serves. This should include a reference to the [crates index spec](https://github.com/rust-lang/rfcs/blob/master/text/2141-alternative-registries.md#registry-index-format-specification) and an example for pulling a starter index repo locally_

_For example_
```
cd ./crate-index
git clone https://github.com/RangerStation/rangerstation-alexandrie-index.git ./
```
And update the git remote url to point to your own repo. *Warning: this command will force push to your repo, overwriting any content that might be there*
```
git remote set-url origin <https url to a repo you own>
git push -f origin master
```

### Starting the container

_notes for `docker run ....` with the flags and shtuff_


### Configuring an alternative registry

_ Notes for configuring the `local` registry_

_ Example for configuring a local index repo as the registry in the `~/.cargo/config` file_
```
[registries.local]
index = "file://localhost/home/ankit/Documents/Dev/crate-index"
```

### Login to your registry

_Visit localhost:3000.....Create a user...Get a token...._

_ In a separate terminal, login to your `local` registry with `cargo login --registry local <token>`

### Publishing a crate

_ Example steps for `cargo new testpackage && cd testpackage && cargo publish --registry local`

## Conclusion